includes/classes/EnqueueScript.php

```diff
@@ -0,0 +1,284 @@
+	 *
+	 * @since x.x.x
+	 */
+	private ?string $localize_script_param_name = null;
+
+	/**
+	 * Script localization parameter data.
+	 *
+	 * @since x.x.x
+	 */
+	private ?array $localize_script_param_data = null;
+
+	/**
+	 * Plugin root directory path.
+	 *
+	 * @since x.x.x
+	 */
+	private string $plugin_dir_path;
+
+	/**
+	 * Plugin root directory URL.
+	 *
+	 * @since x.x.x
+	 */
+	private string $plugin_dir_url;
+
+	/**
+	 * Plugin text domain.
+	 *
+	 * @since x.x.x
+	 */
+	private string $text_domain;
+
+	/**
+	 * EnqueueScript constructor.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param string $script_id   Script ID.
+	 * @param string $script_name Script name.
+	 *
+	 * @throws Exception If script file not found.
+	 */
+	public function __construct( string $script_id, string $script_name ) {
+		$this->plugin_dir_path      = DT_PLUGIN_PATH;
+		$this->plugin_dir_url       = trailingslashit( plugin_dir_url( DT_PLUGIN_FULL_FILE ) );
+		$this->text_domain          = 'distributor';
+		$this->script_id            = $script_id;
+		$this->relative_script_path = 'dist/js/' . $script_name . '.js';
+		$this->absolute_script_path = $this->plugin_dir_path . $this->relative_script_path;
+
+		if ( ! file_exists( $this->absolute_script_path ) ) {
+			throw new Exception( 'Script file not found: ' . $this->absolute_script_path );
+		}
+	}
+
+	/**
+	 * Flag to decide whether load script in footer.
+	 *
+	 * @since x.x.x
+	 */
+	public function load_in_footer(): EnqueueScript {
+		$this->load_script_in_footer = true;
+
+		return $this;
+	}
+
+	/**
+	 * Set script dependencies.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array $script_dependencies Script dependencies.
+	 */
+	public function dependencies( array $script_dependencies ): EnqueueScript {
+		$this->script_dependencies = $script_dependencies;
+
+		return $this;
+	}
+
+	/**
+	 * Register script.
+	 *
+	 * @since x.x.x
+	 */
+	public function register(): EnqueueScript {
+		$script_url   = $this->plugin_dir_url . $this->relative_script_path;
+		$script_asset = $this->get_asset_file_data();
+
+		$this->version = $script_asset['version'];
+
+		wp_register_script(
+			$this->script_id,
+			$script_url,
+			$script_asset['dependencies'],
+			$script_asset['version'],
+			$this->load_script_in_footer
+		);
+
+		if ( $this->register_translations ) {
+			wp_set_script_translations(
+				$this->script_id,
+				$this->text_domain,
+				$this->plugin_dir_path . 'languages'
+			);
+		}
+
+		if ( $this->localize_script_param_data ) {
+			wp_localize_script(
+				$this->script_id,
+				$this->localize_script_param_name,
+				$this->localize_script_param_data
+			);
+		}
+
+		return $this;
+	}
+
+	/**
+	 * This function should be called before enqueue or register function.
+	 *
+	 * @since x.x.x
+	 */
+	public function register_translations(): EnqueueScript {
+		$this->register_translations = true;
+
+		return $this;
+	}
+
+	/**
+	 * This function should be called after enqueue or register function.
```

Comment:
```suggestion
	 * This function should be called after enqueue or register method.
```

However, I am not sure the bit about enqueue is correct. WP places the content of `wp_localize_script()` before the script tag, so I think this will fail if called after enqueue.